skrub/_tabular_learner.py

```diff
@@ -0,0 +1,241 @@
+import sklearn
+from sklearn import ensemble
+from sklearn.base import BaseEstimator
+from sklearn.impute import SimpleImputer
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import OrdinalEncoder
+from sklearn.utils.fixes import parse_version
+
+from ._minhash_encoder import MinHashEncoder
+from ._table_vectorizer import TableVectorizer
+from ._to_categorical import ToCategorical
+
+_HGBT_CLASSES = (
+    ensemble.HistGradientBoostingClassifier,
+    ensemble.HistGradientBoostingRegressor,
+)
+_TREE_ENSEMBLE_CLASSES = (
+    ensemble.HistGradientBoostingClassifier,
+    ensemble.HistGradientBoostingRegressor,
+    ensemble.RandomForestClassifier,
+    ensemble.RandomForestRegressor,
+)
+
+
+def tabular_learner(estimator, n_jobs=None):
+    """Get a simple machine-learning pipeline for tabular data.
+
+    Scikit-learn estimators such as ``LogisticRegression()`` or
+    ``RandomForestClassifier()`` expect their input to be numeric arrays. They
+    do not accept heterogeneous dataframes containing complex data such as
+    datetimes or strings. Moreover, they do not always accept the input to
+    contain missing values. Therefore, some preprocessing must be applied to
+    dataframes before they are passed to an estimator.
```

Comment:
The description on top of the docstring is too long. It should be moved to a "Notes" section, leaving only a quick read before the parameters section.